lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -529,6 +529,31 @@ namespace Js
                 VirtualTableInfo<JavascriptNativeFloatArray>::HasVirtualTable(aValue));
     }
 
+    bool JavascriptArray::IsInlineSegment(SparseArraySegmentBase *seg, JavascriptArray *pArr)
+    {
+        SparseArraySegmentBase* inlineHeadSegment = nullptr;
+        if (JavascriptNativeArray::Is(pArr))
+        {
+            if (JavascriptNativeFloatArray::Is(pArr))
+            {
+                inlineHeadSegment = DetermineInlineHeadSegmentPointer<JavascriptNativeFloatArray, 0, true>((JavascriptNativeFloatArray*)pArr);
+            }
+            else if (JavascriptNativeIntArray::Is(pArr))
```

Comment:
This should be NativeIntArray. FailFast would be good.

---
In reply to: [123650724](https://github.com/Microsoft/ChakraCore/pull/3211#discussion_r123650724) [](ancestors = 123650724)